dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AliasHelper.java

```diff
@@ -0,0 +1,104 @@
+package io.dockstore.webservice.helpers;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.core.Alias;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.core.Workflow;
+import io.dockstore.webservice.core.WorkflowVersion;
+import io.dockstore.webservice.jdbi.WorkflowDAO;
+import io.dockstore.webservice.jdbi.WorkflowVersionDAO;
+import io.dockstore.webservice.resources.AliasableResourceInterface;
+import io.dockstore.webservice.resources.WorkflowResource;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AliasHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AliasHelper.class);
+
+    private AliasHelper() {
+    }
+
+    /**
+     * Finds a workflow and returns the workflow id based on a workflow version id.
+     *
+     * @param workflowDAO Workflow data access object
+     * @param workflowVersionId the id of the workflow version
+     * @return workflow id or throws an exception if the workflow cannot be found
+     */
+    public static long getWorkflowId(WorkflowDAO workflowDAO, long workflowVersionId) {
+        Optional<Long> workflowId = workflowDAO.getWorkflowIdByWorkflowVersionId(workflowVersionId);
+        if (!workflowId.isPresent()) {
+            LOG.error("Could get workflow based on workflow version id " + workflowVersionId);
+            throw new CustomWebApplicationException("Could get workflow based on workflow version id " + workflowVersionId, HttpStatus.SC_NOT_FOUND);
+        }
+        return workflowId.get();
+    }
+
+    /**
+     * Gets the Workflow Version object via a Workflow Version ID
+     * and checks that the Workflow exists and that the
+     * user is authorized to update it
+     * @param workflowResource Workflow resource object
+     * @param workflowDAO Workflow data access object
+     * @param workflowVersionDAO Workflow Version data access object
+     * @param user user authenticated to update the workflow
+     * @param workflowVersionId the id of the workflow version
+     */
+    public static WorkflowVersion getAndCheckWorkflowVersionResource(WorkflowResource workflowResource, WorkflowDAO workflowDAO,
```

Comment:
The methods you are calling on WorkflowResource are in AuthenticatedResourceInterface, so you should pass that instead.

Reply:
done